### DISCUSSION TITLE
How to do a top-down layout but free space goes to the top?

### DISCUSSION THREAD

**Discussion by aurantiaco-sucus** [emilk/egui#4191] [Q&A] [answered]
![demo1](https://github.com/emilk/egui/assets/30406783/ad58eae2-de2a-422b-ad3e-9eacd9058ace)
Like what this diagram shows, I would want to make a layout top-down while letting the free space go to the top. In retained mode UI frameworks it's typically trivial where you put a top-down (expanding) layout in another bottom-top one. But I couldn't quite see how it can be done with egui. I tried applying this very method here, but it doesn't appears to work. Am I doing it wrong? Or is it just plain impossible?

```Rust
ui.with_layout(Layout::bottom_up(Align::Min), |ui| {
        ui.vertical(|ui| {
            ui.label("Hello!");
            ui.label("Hello again!");
        })
    });
```

The snippet above (in an empty window) results in these labels on the top.

**Comment by YgorSouza** [comment, marked as answer]
This is the difficult layout problem mentioned in the readme <https://github.com/emilk/egui/blob/cd1ed733883549bcfe52b009aed8b5c993703314/README.md#layout>. You can use one of the solutions discussed in #3468, storing the size of the top-down UI after you draw it, to calculate the padding to add at the top on the next frame.
